File: sdks/php/src/Types/FactoryContext.php
<?php

namespace Autonoma\Sdk\Types;

class FactoryContext
{
    public function __construct(
        /** @var array<string, array<int, array<string, mixed>>> All refs created so far */
        public readonly array $refs,
        /** The SQL executor (for factories that need direct DB access) */
        public readonly SQLExecutorInterface $executor,
        /** The detected or fallback scope value */
        public readonly string $scenarioName,
        /** Unique ID for this test run */
        public readonly string $testRunId,
    ) {}
}

File: sdks/java/autonoma-sdk/src/main/java/ai/autonoma/sdk/types/FactoryContext.java
package ai.autonoma.sdk.types;

import java.util.List;
import java.util.Map;

/**
 * Context passed to factory create and teardown functions.
 */
public record FactoryContext(
    /** All refs created so far, keyed by model name. */
    Map<String, List<Map<String, Object>>> refs,
    /** The detected or fallback scope value. */
    String scenarioName,
    /** Unique ID for this test run. */
    String testRunId
) {}
